### ISSUE TITLE
Tracking: Android Studio core changes

### ISSUE THREAD

**Issue by skmp** [reicast/reicast-emulator#1120]
Alright, after reviewing the changes in #1099 

I think we need to at least
- [ ] Fixup DYNACALL
- [ ] Rebase from master for the travis PR fixes
- [ ] Open MR against master
- [ ] Fixup build options. Right now -O0 is forced, -fpic is all over, etc.
- [ ] Do we really need to bump to api level 17 and drop xperia support? I'm not against it, just asking.
- [ ] delete `shell/android`

There's probably a few things more things, like Release vs Debug configurations, non-relocable symbols, etc. I don't think any of these are pressing atm.

@LoungeKatt are you up for picking the legwork for this?

**Comment by LoungeKatt** [comment]
I started a fresh branch from master and added the changes from skmp/android-studio, lk/android-studio-core, and lk/android-studio-update, while omitting the reverted changes from lk/android-studio-compat and attempting to reorder and squash when possible.

It does not increase the minimum API, but does remove the original Eclipse folder.

The final product is https://github.com/reicast/reicast-emulator/tree/lk/android-studio, which is based on master and does not rely on any changes to the core or other platforms.

When it comes time to merge this, it can be rebased without conflicts. It can also remain a mirrored branch exclusively for Android Studio. All dependency between the branches has been removed.
